app/models/detection.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# A Detection is a joining record between a Term and a Detector, created when a Detector activates based on some aspect
+# of the Term. This is the record that TACOS found something about this Term.
+#
+# There is a uniqueness constraint on the combination of term_id, detector_id, and detector_version.
+#
+# New records can be created by passing a Term and a Detector object. The model will look up the current detector
+# version, and include that in the record. If this would result in a duplicate record, then the creation process is
+# aborted with no duplicate created.
```

Comment:
Minor suggestion. Rather than saying the process is aborted, I'd say it will throw an exception. One implies maybe we silently continue (but I don't believe we catch the exception to recover cleanly) whereas the other is clear we'll blow up because this should never happen kinda thing. That said, I'm not sure we need this note. I'm not against it, but it feels like something that may easily get outdated as it describes how the model behaves rather than just what it is for. 🤔